readetexts-port.patch

```diff
@@ -155,23 +137,51 @@ index b623d59..5de0af0 100644
 +            # Remove all temporary files
 +            instance_path = os.path.join(self.get_activity_root(), 'instance')
 +            shutil.rmtree(instance_path, ignore_errors=True, onerror=None)
++ 
  
-         self.metadata['activity'] = self.get_bundle_id()
-         self.metadata['mime_type'] = 'application/zip'
-@@ -1146,7 +1130,8 @@ class ReadEtextsActivity(activity.Activity):
-             self.alert(_('Error'), _('You must enter at least one search word.'))
-             self.books_toolbar.search_entry.grab_focus()
-             return
--        f = open('bookcatalog.txt', 'r')
-+        script_dir = os.path.dirname(os.path.realpath(__file__))
-+        f = open(os.path.join(script_dir, 'bookcatalog.txt'), 'r')
-         while f:
-             line = unicode(f.readline(), "iso-8859-1")
-             if not line:
-@@ -1257,30 +1242,13 @@ class ReadEtextsActivity(activity.Activity):
+     def can_close(self):
+         self.close_requested = True
+@@ -1165,13 +1143,13 @@ class ReadEtextsActivity(activity.Activity):
+         tv = selection.get_tree_view()
+         model = tv.get_model()
+         sel = selection.get_selected()
+-        print(sel)
+         if sel:
+             model, _iter = sel
+-            self.selected_title = model.get_value(_iter, COLUMN_TITLE)
+-            self.selected_author = model.get_value(_iter, COLUMN_AUTHOR)
+-            self.selected_path = model.get_value(_iter, COLUMN_PATH)
+-            self.books_toolbar.enable_button(True)
++            if _iter:
++                self.selected_title = model.get_value(_iter, COLUMN_TITLE)
++                self.selected_author = model.get_value(_iter, COLUMN_AUTHOR)
++                self.selected_path = model.get_value(_iter, COLUMN_PATH)
++                self.books_toolbar.enable_button(True)
+ 
+     def find_books(self, search_text):
+         self.clear_downloaded_bytes()
+@@ -1238,13 +1216,13 @@ class ReadEtextsActivity(activity.Activity):
+         logger.debug("Starting download to %s...", path)
+         try:
+             getter.start(path)
++            self.download_content_length = getter.get_content_length()
```

Comment:
Found what I think is the last instance of "move to a separate patch" [1]. For future reference, everything you send to upstream, should be added as a separate patch so that when we update the app version it will be as simple as just removing those patches.

1. https://github.com/sugarlabs/readetexts/pull/19/commits/995428d8f706a4eba992586d818c28e93240bcc2

Reply:
Oki, will remove it too